### PR TITLE
Improve exception messages

### LINKED ISSUE #20951: Rename the exception string parameters for index, fk and key properties
E.g.  index1 -> index1PropertyList

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Part of #7201
Fixes #20951

### CODE REVIEW

File: src/EFCore.Relational/Properties/RelationalStrings.resx [hunk 1]
```diff
@@ -514,7 +514,7 @@
...
   </data>
   <data name="LogMultipleCollectionIncludeWarning" xml:space="preserve">
-    <value>Compiling a query which loads related collections for more than one collection navigation either via 'Include' or through projection but no 'QuerySplittingBehavior' has been configured. By default Entity Framework will use 'QuerySplittingBehavior.SingleQuery' which can potentially result in slow query performance. See https://go.microsoft.com/fwlink/?linkid=2134277 for more information. To identify the query that's triggering this warning call 'ConfigureWarnings(w =&gt; w.Throw(RelationalEventId.MultipleCollectionIncludeWarning))'</value>
+    <value>Compiling a query which loads related collections for more than one collection navigation either via 'Include' or through projection but no 'QuerySplittingBehavior' has been configured. By default Entity Framework will use 'QuerySplittingBehavior.SingleQuery' which can potentially result in slow query performance. See https://go.microsoft.com/fwlink/?linkid=2134277 for more information. To identify the query that's triggering this warning call 'ConfigureWarnings(w =&gt; w.Throw(RelationalEventId.MultipleCollectionIncludeWarning)).'</value>
```
Comment (smitpatel) [inline src/EFCore.Relational/Properties/RelationalStrings.resx:517]:
```suggestion
    <value>Compiling a query which loads related collections for more than one collection navigation either via 'Include' or through projection but no 'QuerySplittingBehavior' has been configured. By default Entity Framework will use 'QuerySplittingBehavior.SingleQuery' which can potentially result in slow query performance. See https://go.microsoft.com/fwlink/?linkid=2134277 for more information. To identify the query that's triggering this warning call 'ConfigureWarnings(w =&gt; w.Throw(RelationalEventId.MultipleCollectionIncludeWarning))'.</value>
```

File: src/EFCore/Properties/CoreStrings.resx [hunk 2]
```diff
@@ -448,31 +448,31 @@
     <value>The [ForeignKey] attribute for the navigation '{navigation}' cannot be specified on the entity type '{principalType}' since it represents a one-to-many relationship. Move the [ForeignKey] attribute to a property on '{dependentType}'.</value>
   </data>
   <data name="FkAttributeOnPropertyNavigationMismatch" xml:space="preserve">
-    <value>The [ForeignKey] attributes on property '{property}' and navigation '{navigation}' in entity type '{entityType}' do not point at each other. The value of the [ForeignKey] attribute on the property should be navigation name, and the value of the [ForeignKey] attribute on the navigation should be the foreign key property name.</value>
+    <value>The [ForeignKey] attributes on property '{property}' and navigation '{navigation}' in entity type '{entityType}' do not point at each other. Make sure the value of the [ForeignKey] attribute on the property mathes the navigation name, and the value of the [ForeignKey] attribute on the navigation mathes the foreign key property name or remove one of them.</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:451]: mathes

File: src/EFCore/Properties/CoreStrings.resx [hunk 3]
```diff
@@ -448,31 +448,31 @@
...
-    <value>The number of properties specified for the foreign key {foreignKeyProperties} on entity type '{dependentType}' does not match the number of properties in the principal key {principalKey} on entity type '{principalType}'.</value>
+    <value>The number of properties specified for the foreign key {foreignKeyProperties} on entity type '{dependentType}' does not match the number of properties in the principal key {principalKeyProperties} on entity type '{principalType}'.</value>
   </data>
   <data name="ForeignKeyInUseSkipNavigation" xml:space="preserve">
-    <value>Cannot remove the foreign key {foreigKey} from entity type '{entityType}' because it is referenced by a skip navigation '{navigation}' on entity type '{navigationEntityType}'. All referencing skip navigation must be removed before the referenced foreign key can be removed.</value>
+    <value>Cannot remove the foreign key {foreigKeyProperties} from entity type '{entityType}' because it is referenced by a skip navigation '{navigation}' on entity type '{navigationEntityType}'. All referencing skip navigation must be removed before the referenced foreign key can be removed.</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:460]: foreigKeyProperties

Comment (smitpatel) [reply src/EFCore/Properties/CoreStrings.resx:460]: > All referencing skip navigation
plural?

File: src/EFCore/Properties/CoreStrings.resx [hunk 4]
```diff
@@ -589,16 +590,16 @@
...
   <data name="InvalidSetType" xml:space="preserve">
     <value>Cannot create a DbSet for '{typeName}' because this type is not included in the model for the context.</value>
   </data>
   <data name="InvalidSetTypeOwned" xml:space="preserve">
-    <value>Cannot create a DbSet for '{typeName}' because it is configured as an owned entity type and must be accessed through its owning entity type.</value>
+    <value>Cannot create a DbSet for '{typeName}' because it is configured as an owned entity type and must be accessed through its owning entity type '{ownerType}'.</value>
   </data>
   <data name="InvalidSetTypeWeak" xml:space="preserve">
-    <value>Cannot create a DbSet for '{typeName}' because it is mapped to multiple entity types and should be accessed through the owning entities.</value>
+    <value>Cannot create a DbSet for '{typeName}' because it is mapped to multiple entity types with defininf navigations and should be accessed through the owning entities.</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:602]: defininf

File: src/EFCore/Properties/CoreStrings.resx [hunk 5]
```diff
@@ -972,46 +973,46 @@
     <value>The filters '{filter1}' and '{filter2}' have both been configured on the same included navigation. Only one unique filter per navigation is allowed. For more information on including related data, see http://go.microsoft.com/fwlink/?LinkID=746393.</value>
   </data>
   <data name="MultipleNavigationsSameFk" xml:space="preserve">
-    <value>There are multiple navigations in entity type '{entityType}' which are pointing to same set of properties using a [ForeignKey] attribute: '{propertyList}'</value>
+    <value>There are multiple navigations in entity type '{entityType}' which are pointing to same set of properties '{properties}' using a [ForeignKey] attribute: {navigations}.</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:976]: Would this properties require quotes around?

File: src/EFCore/Properties/CoreStrings.resx [hunk 6]
```diff
@@ -972,46 +973,46 @@
...
   <data name="NamedIndexWrongType" xml:space="preserve">
     <value>The index with name {indexName} cannot be removed from the entity type '{entityType}' because no such index exists on that entity type.</value>
   </data>
   <data name="NavigationArray" xml:space="preserve">
-    <value>The type of navigation '{1_entityType}.{0_navigation}' is '{foundType}' which is an array type. Collection navigation properties cannot be arrays.</value>
+    <value>The type of navigation '{1_entityType}.{0_navigation}' is '{foundType}' which is an array type. Collection navigations cannot be arrays.</value>
   </data>
   <data name="NavigationBadType" xml:space="preserve">
-    <value>The type of navigation '{1_entityType}.{0_navigation}' is '{foundType}' which does not implement ICollection&lt;{targetType}&gt;. Collection navigation properties must implement ICollection&lt;&gt; of the target type.</value>
+    <value>The type of navigation '{1_entityType}.{0_navigation}' is '{foundType}' which does not implement ICollection&lt;{targetType}&gt;. Collection navigations must implement ICollection&lt;&gt; of the target type.</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:1000]: 'ICollection&lt;&gt;' ?

File: src/EFCore/Properties/CoreStrings.resx (hunk 5 above)
Comment (AndriySvyryd) [reply src/EFCore/Properties/CoreStrings.resx:976]: Yes, this one just contains the string specified in [ForeignKey]

File: src/EFCore/Properties/CoreStrings.resx [hunk 7]
```diff
@@ -997,7 +997,7 @@
     <value>The type of navigation '{1_entityType}.{0_navigation}' is '{foundType}' which is an array type. Collection navigations cannot be arrays.</value>
   </data>
   <data name="NavigationBadType" xml:space="preserve">
-    <value>The type of navigation '{1_entityType}.{0_navigation}' is '{foundType}' which does not implement ICollection&lt;{targetType}&gt;. Collection navigations must implement ICollection&lt;&gt; of the target type.</value>
+    <value>The type of navigation '{1_entityType}.{0_navigation}' is '{foundType}' which does not implement 'ICollection&lt;{targetType}&gt;'. Collection navigations must implement 'ICollection&lt;&gt;' of the target type.</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:1000]: There are few more like this in this file. e.g. NavigationCannotCreateType

**Review by smitpatel — APPROVED** [review]

**Pull request merged**